### DISCUSSION TITLE
Perform http request X times per hour to fetch data from an external API?

### DISCUSSION THREAD

**Discussion by stanifert** [supabase/supabase#10411] [Questions] [answered]
I would like to set up a Supabase database that pings an external api X times per hour, and inserts objects received in that response into a table. The client application would then request data from that table via Supabase's API. I am unsure how to go about this as I'm pretty new to the world of web/server development, but have two years of experience as an iOS dev using Swift. Unfortunately, I have little familiarity with other languages.

I found [this discussion](https://github.com/supabase/supabase/discussions/7646) that is very similar to what I am looking for, but it seems that Supabase functions have since been released. So what would be the best way to go about this now? Would I be able to setup this functionality purely via the Supabase dashboard (and perhaps some code blocks)? Or would I need to setup a local env on my machine and write most of it myself?

I understand my question is pretty broad, but if you could point me in the right direction that would be great! Thanks!

**Comment by GaryAustin1** [comment, marked as answer]
https://supabase.com/blog/postgres-as-a-cron-server 
Has a little more detail on the cron and pg_http method.

With edge functions you will still need something to call them every hour.
If you can do what you need in the database function that is the simplest method.  You could generate the code with the SQL editor in the dashboard.

**Reply by stanifert** [reply]
Thank you!
